StreamVideoTests/TestUtils/AssertDelay.swift

```diff
@@ -0,0 +1,15 @@
+//
+// Copyright © 2023 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+import XCTest
+
+@MainActor
+func XCTAssertWithDelay(
+    _ expression: @autoclosure () throws -> Bool,
+    nanoseconds: UInt64 = 500_000_000
+) async throws {
+    try await Task.sleep(nanoseconds: nanoseconds)
```

Comment:
Rather than using `sleep` here you can achieve the same result by using expectations.

```suggestion
let waitExpectation = expectation(description: "Wait for some time.")
waitExpectation.isInverted = true
wait(for: [waitExpectation], timeout: defaultTimeout)
try expression()
```